File: src/AgentOrchestrator/AgentHQDemo.Api/Services/LogSanitizer.cs
namespace AgentHQDemo.Api.Services;

/// <summary>
/// Helpers for writing user-supplied values to logs safely.
/// </summary>
internal static class LogSanitizer
{
    /// <summary>
    /// Removes line breaks from a user-supplied value so it cannot be used to
    /// forge additional log entries (CWE-117).
    /// </summary>
    public static string Sanitize(string? value)
    {
        if (string.IsNullOrEmpty(value))
        {
            return string.Empty;
        }

        return value.Replace("\r", string.Empty, StringComparison.Ordinal)
                    .Replace("\n", string.Empty, StringComparison.Ordinal);
    }
}

File: src/AgentOrchestrator-python/app/log_sanitizer.py
"""Helpers for writing user-supplied values to logs safely.

Mirrors ``AgentHQDemo.Api/Services/LogSanitizer.cs``.
"""


def sanitize_for_log(value: str | None) -> str:
    """Removes line breaks from a user-supplied value so it cannot be used to
    forge additional log entries (CWE-117).
    """
    if not value:
        return ""

    return value.replace("\r", "").replace("\n", "")
